src/config.ts

```diff
@@ -24,16 +27,62 @@ function configOrder(obj: configObjects): configObjects {
   return obj
 }
 
-export function getDefaultShell(): string {
+// finds the default shell start commmand
+async function getDefaultShell(): Promise<string> {
+  let shellStartCommand
+  if (process.platform === "win32") {
+    // Windows
+    try {
+      shellStartCommand = await which("pwsh.exe")
+      return shellStartCommand
+    } catch (e1) {
+      try {
+        shellStartCommand = await which("powershell.exe")
+        return shellStartCommand
+      } catch (e2) {
+        shellStartCommand = process.env.COMSPEC || "cmd.exe"
+        return shellStartCommand
+      }
+    }
+  } else {
+    // Unix
+    shellStartCommand = process.env.SHELL || "/bin/sh"
+    return shellStartCommand
+  }
+}
+
+// The shell command used in case getDefaultShell does not find a prefered shell
+function getFallbackShell() {
   return process.platform === "win32" ? process.env.COMSPEC || "cmd.exe" : process.env.SHELL || "/bin/sh"
 }
 
+// set start command asyncronously
+export async function setShellStartCommand(disposables: CompositeDisposable) {
+  if (atom.config.get("terminal.autoShell")) {
+    const shellStartCommand = await getDefaultShell()
+    atom.config.set("terminal.shell", shellStartCommand)
+  }
+  return disposables.add(
+    // an observre that checks if command has been edited manually, if so it will turn autoShell off
+    atom.config.observe("terminal.shell", () => {
+      atom.config.set("terminal.autoShell", false)
+    })
+  )
+}
+
 export const config = configOrder({
+  autoShell: {
+    title: "Automatic shell detection",
+    description:
+      "Automatically detect the prefered shell start command in the next Atom restart. (it will switched off if you edit `shell` option manually.",
+    type: "boolean",
+    default: true,
+  },
```

Comment:
For now, we need to use another "watcher config". We can hide this watcher by storing it in the LocalStorage of the Chrome engine.

I don't think having this "autoShell" in the configuration help much. Since this is going to set "a default value", and once the user writes something manually, it gets turned off. 

In conclusion, I prefer to hide this.

Reply:
I like the idea of local storage instead of a "watcher config".

Check if a flag (`localStorage.terminalShellSet`) is set and if it is not then run setShellStartCommand and set the `shell` config on activate.

Reply:
I added this in https://github.com/atom-ide-community/terminal/pull/24/commits/b71d8438f0f2615b3bcc83282e92235252640e76